src/js/tech/html5.js

```diff
@@ -119,6 +121,125 @@ class Html5 extends Tech {
     super.dispose();
   }
 
+  /**
+   * A special function to trigger source set in a way that
+   * will allow player to re-trigger if player/html5 are not ready
+   * yet.
+   *
+   * @fires Tech#sourceset
+   */
+  triggerSourceSet_() {
```

Comment:
should we trigger `sourceset` asynchronously? Also, I'm thinking maybe moving this method to the `tech` object so that other techs could just call `this.triggerSourceSet_()` whenever necessary if/when they build out support for this.

Reply:
I could see moving triggerSourceSet out and I think that makes sense

Reply:
I am not sure if that would work, I guess we could include the source that was set in the event and it would be fine. But at one point it was said that we want something like this to work:
```this.on('sourceset', (e) => {
  console.log(player.currentSource())
});

// should log `{type: 'fake1', src: 'test1'}` above
player.src({type: 'fake1', src: 'test1'});

// should log `{type: 'fake2', src: 'test2'}` above
player.src({type: 'fake2', src: 'test2'});

// should log `{type: 'fake3', src: 'test3'}` above
player.src({type: 'fake3', src: 'test3'});

```


or maybe this

```this.one('sourceset', (e) => {
  // should be fake1, test1
  console.log(player.currentSource());

  this.one('sourceset', () => {
     // should be fake2, test2
     console.log(player.currentSource());
  })
  player.src({type: 'fake2', src: 'test2'});
});

player.src({type: 'fake1', src: 'test2'});
```

If it was async I am not sure that it would